From 6faf5e963eadeab8b7d47d5b3c14f06e0ae1da01 Mon Sep 17 00:00:00 2001
From: Hans Verkuil <hverkuil@xs4all.nl>
Date: Sat, 12 Apr 2025 12:30:13 +0200
Subject: [PATCH] meson.build: fix arm _TIME_BITS=64 error

Undefine _TIME_BITS to avoid this error on 32-bit arm:

/usr/include/features-time64.h:26:5: error: #error "_TIME_BITS=64 is allowed only with _FILE_OFFSET_BITS=64"

Signed-off-by: Hans Verkuil <hverkuil@xs4all.nl>
Upstream: https://git.linuxtv.org/v4l-utils.git/commit/?id=d517cfdcdc16533ab7e06e97c07ca089cf261aef
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 269a9da7..31927cda 100644
--- a/meson.build
+++ b/meson.build
@@ -53,8 +53,12 @@ v4l2_wrapper_args = [
     # As the library needs to provide both 32-bit and 64-bit versions
     # of file operations, disable transparent large file support (fixes
     # 'Error: symbol `open64/mmap64' is already defined' compile failure
-    # otherwise)
+    # otherwise).
+    #
+    # Also disable _TIME_BITS=64 since this is allowed only with
+    # _FILE_OFFSET_BITS=64, which is now 32.
     '-U_FILE_OFFSET_BITS',
+    '-U_TIME_BITS',
     '-D_FILE_OFFSET_BITS=32',
     '-D_LARGEFILE64_SOURCE',
 ]
-- 
2.49.0

